spec/unit/typesafe/config/config_value_spec.rb

```diff
@@ -641,3 +647,290 @@ def n(v)
+  end
+
+  specify "works with value depth 2 from empty" do
+    v = ConfigValueFactory.from_any_ref(42)
+    config = ConfigFactory.empty.with_value("a.b", v)
+
+    expect(config).to eq(TestUtils.parse_config("a.b=42"))
+    expect(v).to eq(config.get_value("a.b"))
+  end
+
+  specify "works with value depth 3 from empty" do
+    v = ConfigValueFactory.from_any_ref(42)
+    config = ConfigFactory.empty.with_value("a.b.c", v)
+
+    expect(config).to eq(TestUtils.parse_config("a.b.c=42"))
+    expect(v).to eq(config.get_value("a.b.c"))
+  end
+
+  specify "with value depth 1 overwrites existing" do
+    v = ConfigValueFactory.from_any_ref(47)
+    old = v.at_path("a")
+    config = old.with_value("a", ConfigValueFactory.from_any_ref(42))
+
+    expect(config).to eq(TestUtils.parse_config("a=42"))
+    expect(config.get_int("a")).to eq(42)
+  end
+
+  specify "with value depth 2 overwrites existing" do
+    v = ConfigValueFactory.from_any_ref(47)
+    old = v.at_path("a.b")
+    config = old.with_value("a.b", ConfigValueFactory.from_any_ref(42))
+
+    expect(config).to eq(TestUtils.parse_config("a.b=42"))
+    expect(config.get_int("a.b")).to eq(42)
+  end
+
+  specify "with value inside existing object" do
+    v = ConfigValueFactory.from_any_ref(47)
+    old = v.at_path("a.c")
+    config = old.with_value("a.b", ConfigValueFactory.from_any_ref(42))
+
+    expect(config).to eq(TestUtils.parse_config("a.b=42,a.c=47"))
+    expect(config.get_int("a.b")).to eq(42)
+    expect(config.get_int("a.c")).to eq(47)
+  end
+
+  specify "with value build complex config" do
+    v1 = ConfigValueFactory.from_any_ref(1)
+    v2 = ConfigValueFactory.from_any_ref(2)
+    v3 = ConfigValueFactory.from_any_ref(3)
+    v4 = ConfigValueFactory.from_any_ref(4)
+
+    config = ConfigFactory.empty.with_value("a", v1)
+      .with_value("b.c", v2)
+      .with_value("b.d", v3)
+      .with_value("x.y.z", v4)
+
+    expect(config).to eq(TestUtils.parse_config("a=1,b.c=2,b.d=3,x.y.z=4"))
+  end
+end
+
+describe "#render" do
+  context "has newlines in description" do
+    v = ConfigValueFactory.from_any_ref(89, "this is a description\nwith some\nnewlines")
+
+    list = SimpleConfigList.new(SimpleConfigOrigin.new_simple("\n5\n6\n7\n"), [v])
+
+    conf = ConfigFactory.empty.with_value("bar", list)
+
+    rendered = conf.root.render
+
+    specify "rendered config should have all the lines that were added, with newlines" do
+      expect(rendered).to include("is a description\n")
+      expect(rendered).to include("with some\n")
+      expect(rendered).to include("newlines\n")
+      expect(rendered).to include("#\n")
+      expect(rendered).to include("5\n")
+      expect(rendered).to include("6\n")
+      expect(rendered).to include("7\n")
+    end
+
+    specify "the rendered config should give back the original config" do
+      skip "unknown bug related to comments that will be affected by upstream changes to comment parsing"
```

Comment:
We might want to put a TODO here so we make sure to come back and stop skipping this test once the parser gets refactored